dotnet-desktop-guide/net/winforms/whats-new/net60.md

```diff
@@ -0,0 +1,162 @@
+ms.date: 10/28/2021
+ms.topic: conceptual
+---
+
+# What's new for .NET 6 (Windows Forms .NET)
+
+Windows Forms for .NET 6 adds the following features and enhancements over .NET Framework.
+
+There are a few breaking changes you should be aware of when migrating from .NET Framework to .NET 6. For more information, see [Breaking changes in Windows Forms](/dotnet/core/compatibility/winforms).
+
+## New application default font
+
+[`Control.DefaultFont`](https://docs.microsoft.com/dotnet/api/system.windows.forms.control.defaultfont?view=windowsdesktop-5.0) in .NET Framework is set to Microsoft Sans Serif, 8.25pt. In .NET Core 3.0 the value of the default font has been changed to Segoe UI, 9pt align with [Windows user experience (UX) guidelines](https://docs.microsoft.com/windows/win32/uxguide/vis-fonts#fonts-and-colors). This change, however, made harder for some customers to migrate their large applications with pixel-perfect layouts.
+
+To make it easier to migrate those pixel-perfect appslications we added a new [`Application.SetDefaultFont`](https://docs.microsoft.com/dotnet/api/system.windows.forms.application.setdefaultfont) API to our toolkit:
+
+```csharp
+[STAThread]
+static void Main()
+{
+    Application.EnableVisualStyles();
+    Application.SetHighDpiMode(HighDpiMode.PerMonitorV2);
+
+    Application.SetDefaultFont(new Font(new FontFamily("Microsoft Sans Serif"), 8.25f));
+
+    Application.Run(new Form1());
+}
+```
+
+## New application bootstrap
+
+The change of the default font has also meant the designer was no longer able to provide a true WYSIWYG experience, because Visual Studio process is run under .NET Framework 4.7.2 and uses the old default font, whereas a .NET application at runtime uses the new font.
+
+To make the designer the default font aware, and to simplify the `Main` method we have re-designed [the application bootstrap](https://github.com/dotnet/designs/blob/main/accepted/2021/winforms/streamline-application-bootstrap.md):
+
+```csharp
+class Program
+{
+    [STAThread]
+    static void Main()
+    {
+        ApplicationConfiguration.Initialize();
+        Application.Run(new Form1());
+    }
+}
+```
+
+However, there is more a play than immediately meets the eye.
+`ApplicationConfiguration.Initialize()` is a source generated API that behind the scenes emits the following calls:
+```csharp
+Application.EnableVisualStyles();
+Application.SetCompatibleTextRenderingDefault(false);
+Application.SetDefaultFont(new Font(...));
+Application.SetHighDpiMode(HighDpiMode.SystemAware);
+```
+The parameters of these calls are configurable via [MSBuild properties](https://docs.microsoft.com/dotnet/core/project-sdk/msbuild-props-desktop#windows-forms-settings) in csproj or props files.
+The Windows Forms designer in Visual Studio 2022 is also aware of these properties (for now it only reads the default font), and can show you your application as it would look at runtime:
+
+<TODO: image goes here>
```

Comment:
@adegeo this section's wording is a compilation from my blog and and @richlander's blog (yet to be published). Feel free to reword as you see necessary.
I added an image to this commit that's being used in the blogs (appbootstrap.png). I don't know whether we want to add it to the docs.

Reply:
https://devblogs.microsoft.com/dotnet/announcing-net-6/

Reply:
https://devblogs.microsoft.com/dotnet/whats-new-for-visual-basic-in-visual-studio-2022/

Reply:
@RussKie Can you clarify something? I'm working on this section and the way this is worded makes it sound like the default font change that was previously introduced was reverted and now set via the `SetDefaultFont` method.  So if someone removes the call to that method, the default font remains the same as it was .NET Framework.

Correct?

Reply:
> @RussKie Igor Velikorossov FTE Can you clarify something? I'm working on this section and the way this is worded makes it sound like the default font change that was previously introduced was reverted and now set via the SetDefaultFont method. So if someone removes the call to that method, the default font remains the same as it was .NET Framework.
>
>Correct?

Actually no, not quite correct. The Default Font changed early on, but lots of developers who were porting their applications from Framework, where they were laying out based on the old font found that with the new default font it looked ugly. Either they had to change the font on every form in their application or realign their layout. As a result we decided to ADD the new Application wide default font so that if a developer wanted to use the old defualt they could explicitly state that's what they wanted application wide. Make sense?